lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +468,52 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+    def get_chunk(self, trans, dataset, offset=0, ck_size=None):
+        index_file = dataset.metadata.bam_index
+        with pysam.AlignmentFile(dataset.file_name, "rb", index_filename=index_file.file_name) as bamfile:
+            ck_size = 1000  # 1000 lines
+            ck_data = ""
+            line_number = 0
+            if offset == 0:
+                ck_data = bamfile.text
+            for line_number, alignment,  in enumerate(bamfile):
+                if line_number > offset and line_number <= (offset + ck_size):
```

Comment:
Actually, we can switch this around and break here if the `line_number` exceeds `ck_size`:
```
                if line_number > ck_size:
                    break
```